torch/distributed/tensor/parallel/fsdp.py

```diff
@@ -29,6 +29,12 @@ def enable_2d_with_fsdp() -> bool:
 
     torch._C._log_api_usage_once("torch.distributed.tensor.parallel.enable_2d_with_fsdp")
 
+    warnings.warn(
+        "`enable_2d_with_fsdp` will be deprecated soon.",
+        "Please avoid using it.",
```

Comment:
I think we shouldn't throw warning until we have the new way ready

Reply:
> I think we shouldn't throw warning until we have the new way ready

Converting it back to draft and will re-publish later.